File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if(head == null) return head;
        ListNode node = head, current = head;
        while(node != null && current != null){
            if(current.val != node.val){
                node.next = current;
                node = current;
            }
            current = current.next;
        }
        node.next = null;
        return head;
    }
}

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    if(head === null) return head;
    let node = head;
    let current = head;
    while(current !== null && node !== null){
        if(node.val !== current.val){
            node.next = current;
            node = current;
        }
        current = current.next;
    }
    node.next=null;
    return head;
};